editor/src/components/canvas/canvas-strategies/strategies/resize-grid-strategy.ts

```diff
@@ -169,6 +199,64 @@ export const resizeGridStrategy: CanvasStrategyFactory = (
   }
 }
 
+type ExpandedGridDimension = GridDimension & {
+  repeated: boolean
+  originalIndex: number
+}
+
+// Reconstruct the final, resized dimensions trying to preserve as much as possible the original CSS.
+// 1. Determine the original index of the targeted dimension, as the one that is found in the original values array
+// 2. Extract the left and right portions of the dimensions from the original values, as they did not change
+// 3. Calculate the "mid chunk" portion of the dimensions, which was affected by the resize.
+//    If the target is part of a repeated dimension, derive the final repeat so that the repeat statement itself
+//    is not denormalized but updated to repeat with the new dimensions
+// 4. Finally build the new dimensions as the concatenation of the left chunk, mid chunk, and right chunk.
+function buildResizedDimensionsArray(params: {
+  expandedOriginalValues: ExpandedGridDimension[]
+  originalValues: GridDimension[]
+  resizedValues: GridDimension[]
+  resizedIndex: number
+}): GridDimension[] {
+  // 1.
+  const targetIndex = params.expandedOriginalValues[params.resizedIndex].originalIndex
+  const targetDimension = params.originalValues[targetIndex]
+
+  // 2.
+  const leftChunk = params.originalValues.slice(0, targetIndex)
+  const rightChunk = params.originalValues.slice(targetIndex + 1)
+
+  // 3.
+  // midChunkStart is the index in the expanded values matching the first occurrence of the target index in the
+  // expanded values.
+  const midChunkStart = params.expandedOriginalValues.findIndex(
+    (c) => c.originalIndex === targetIndex,
+  )
+  // midChunkEnd is the start + the number of repeated elements for this `repeat` function
+  const midChunkEnd =
+    midChunkStart +
+    params.expandedOriginalValues.filter((c) => c.originalIndex === targetIndex).length
```

Comment:
I think the name `midChunkEnd` is now incorrect if this is a `repeat`, because of the case e.g. `repeat(2, 5px, 10px)`. It might be clearer to have `const midChunkSize = params.expandedOriginalValues.filter((c) => c.originalIndex === targetIndex).length` and then only introduce `midChunkEnd` in the `else` statement

Reply:
[`32c1467` (#6380)](https://github.com/concrete-utopia/utopia/pull/6380/commits/32c1467d962b501ec45a63871df1f4ebf5088d49)